### PR TITLE
P8/P9 - Adding Positional Scoring

### PR DESCRIPTION
**Pull request by trdougherty**
* Switching to positional scoring
* Introducing new drawdown softening
* New consistency penalties for positional criteria
* Update to challenge period
* Cleaning up minerstatistics
* Rebuilding incentive_review to use outputs of minerstatistics
* Internal code cleaning, removing old functions / files

### Description
[Provide a brief description of the changes introduced by this pull request.]

### Related Issues (JIRA)
[Reference any related issues or tasks that this pull request addresses or closes.]

### Checklist
- [ ] I have tested my changes on testnet.
- [ ] I have updated any necessary documentation.
- [ ] I have added unit tests for my changes (if applicable).
- [ ] If there are breaking changes for validators, I have (or will) notify the community in Discord of the release.

### Reviewer Instructions
[Provide any specific instructions or areas you would like the reviewer to focus on.]

### Definition of Done
- [ ] Code has been reviewed.
- [ ] All checks and tests pass.
- [ ] Documentation is up to date.
- [ ] Approved by at least one reviewer.

### Checklist (for the reviewer)
- [ ] Code follows project conventions.
- [ ] Code is well-documented.
- [ ] Changes are necessary and align with the project's goals.
- [ ] No breaking changes introduced.

### Optional: Deploy Notes
[Any instructions or notes related to deployment, if applicable.]

/cc @mention_reviewer

### CODE REVIEW

File: vali_objects/utils/subtensor_weight_setter.py
Comment (trdougherty) [inline vali_objects/utils/subtensor_weight_setter.py:84]: @jbonilla-tao @suntao99 this is what's driving my request for purge, as we have a number of inactive hotkeys which should be purged before our eliminated miners

File: vali_objects/utils/subtensor_weight_setter.py
Comment (suntao99) [inline vali_objects/utils/subtensor_weight_setter.py:114]: Is this left here intentionally?

File: vali_objects/utils/position_penalties.py
Comment (suntao99) [inline vali_objects/utils/position_penalties.py:126]: Nice usage of list comprehension.

File: vali_objects/utils/subtensor_weight_setter.py
Comment (suntao99) [inline vali_objects/utils/subtensor_weight_setter.py:45]: Is this left here intentionally?

**Review by suntao99 — APPROVED** [review]
This PR looks good! There's a few comments but not blocking. However, there should be another set of eyes on this who are closer to the project source code who can look at the business logic. Also, this PR is huge, we might want to break it down in smaller PRs for easier review. It might be because it's not rebased against the latest and greatest? the "main" branch in this case. I would check to see if it's properly rebased.

Typically, we wouldn't want to PR directly into main. I recommend setting up a staging branch and PR into that instead. And when you're ready to release into production, create a release PR from staging into main with proper release tags. I can walk you through this if necessary. 

Approved with comments! Non blocking. Need someone else familiar with source code for second set of eyes.

File: vali_objects/utils/subtensor_weight_setter.py
Comment (trdougherty) [reply vali_objects/utils/subtensor_weight_setter.py:84]: @sli-tao this is what I was talking about which I think is driving the new challengeperiod logic change

File: vali_objects/utils/elimination_manager.py
Comment (jbonilla-tao) [inline vali_objects/utils/elimination_manager.py:76]: can just do this once before the loop over all hotkeys. reading from disk is expensive.

File: vali_objects/utils/subtensor_weight_setter.py
Comment (jbonilla-tao) [reply vali_objects/utils/subtensor_weight_setter.py:84]: This seems redundant if we add purging in the elimination manager

File: vali_objects/utils/subtensor_weight_setter.py
Comment (trdougherty) [reply vali_objects/utils/subtensor_weight_setter.py:114]: @jbonilla-tao @sli-tao do we think it's a good idea / necessary to coerce syncing on the positions and ledgers like this? How long does it take for the ledger to generate on the first position

File: vali_objects/utils/elimination_manager.py
Comment (trdougherty) [reply vali_objects/utils/elimination_manager.py]: updated

File: vali_objects/utils/subtensor_weight_setter.py
Comment (trdougherty) [reply vali_objects/utils/subtensor_weight_setter.py]: sounds good. Updated this to remove and purge any instances of eliminated miners from challenge period, so we don't need to reference it in the main weight setter

File: vali_objects/utils/subtensor_weight_setter.py
Comment (jbonilla-tao) [reply vali_objects/utils/subtensor_weight_setter.py:110]: On average the time delta should be ~20 minutes between the first position and the initial perf ledger. I think upperbound could be 1 hour. I don't think forced sync is possible or needed. Can simply wait until the ledger exists.

File: validation/tmp/b177e1f6-7115-40c0-9d9e-da1ac9a83977
Comment (sli-tao) [inline validation/tmp/b177e1f6-7115-40c0-9d9e-da1ac9a83977:1]: remove with
```
git rm --cached path/to/generated-file
git commit --amend --no-edit
git push --force
```

File: vali_objects/utils/subtensor_weight_setter.py
Comment (sli-tao) [reply vali_objects/utils/subtensor_weight_setter.py:110]: If the ledger and positions are out of sync would that generate incorrect miner statistics?

File: validation/tmp/b177e1f6-7115-40c0-9d9e-da1ac9a83977
Comment (trdougherty) [reply validation/tmp/b177e1f6-7115-40c0-9d9e-da1ac9a83977:1]: oh good catch

**Comment by sli-tao** [comment]
PR looks good, nice tests! Thanks for fixing up the linting!

**Review by sli-tao — APPROVED** [review]
Don't forget to set TARGET_MS before merging

File: vali_objects/utils/subtensor_weight_setter.py
Comment (trdougherty) [reply vali_objects/utils/subtensor_weight_setter.py:110]: think we're ok. scoring should skip over miners without both

**Pull request merged**
